Sources/SwiftProtobuf/DoubleParser.swift

```diff
@@ -0,0 +1,63 @@
+// Sources/SwiftProtobuf/DoubleParser.swift - Generally useful mathematical functions
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See LICENSE.txt for license information:
+// https://github.com/apple/swift-protobuf/blob/master/LICENSE.txt
+//
+// -----------------------------------------------------------------------------
+///
+/// Numeric parsing helper for float and double strings
+///
+// -----------------------------------------------------------------------------
+
+import Foundation
+
+/// Support parsing float/double values from UTF-8
+internal class DoubleParser {
+    #if swift(>=4.1)
+      private var work =
+          UnsafeMutableRawBufferPointer.allocate(byteCount: 128,
```

Comment:
Document how/why 128 was chosen?

Reply:
Done.

Reply:
After looking at this, I added another sanity check to guard against this being called with a zero-length (or negative-length) buffer.